src/Interpreters/ClusterDiscovery.cpp

```diff
@@ -293,21 +356,20 @@ static bool contains(const Strings & list, const String & value)
 
 /// Reads data from zookeeper and tries to update cluster.
 /// Returns true on success (or no update required).
-bool ClusterDiscovery::updateCluster(ClusterInfo & cluster_info)
+bool ClusterDiscovery::upsertCluster(ClusterInfo & cluster_info)
```

Comment:
Can we also update a comment that this method can also create new cluster or remove it